### PR TITLE
psm: use double instead of float

### PR DESCRIPTION
**Pull request by gadfort**
Changes:
- switch Voltages to doubles to avoid some casting during solution generation
- use long-double when generating conductance matrix to avoid rounding error

Fixes:
- numerical instability in CI (hopefully).

### CODE REVIEW

File: src/psm/src/ir_solver.cpp [hunk 1]
```diff
@@ -807,7 +807,7 @@ void IRSolver::buildCondMatrixAndVoltages(
       J[node_idx] = find_node->second;
     }
 
-    Connection::Conductance node_cond = 0.0;
+    long double node_cond = 0.0;
```
Comment (QuantamHD) [inline src/psm/src/ir_solver.cpp:810]: Maybe we should add another using declartation called `Connection::ConductanceHighPrecision`

Comment (maliberty) [reply src/psm/src/ir_solver.cpp:810]: long double may have performance implications (using x87 and not SSE).  Is this strictly necessary?

Comment (gadfort) [reply src/psm/src/ir_solver.cpp:810]: I think I found a seconds stop that could avoid this... so I converted to draft until I can verify.

**Comment by QuantamHD** [comment]
LGTM. Sort looks good

**Review by QuantamHD — APPROVED** [review]

File: src/psm/src/ir_solver.h [hunk 2]
```diff
@@ -187,7 +187,8 @@ class IRSolver
   bool wasNodeVisited(const std::unique_ptr<Node>& node) const;
   bool wasNodeVisited(const Node* node) const;
 
-  std::map<Node*, std::set<Connection*>> getNodeConnectionMap(
+  std::map<Node*, std::set<Connection*, Connection::Compare>>
```
Comment (maliberty) [inline src/psm/src/ir_solver.h:190]: `std::set<Connection*, Connection::Compare>` get repeated alot, perhaps `using ConnectionSet = std::set<Connection*, Connection::Compare>` ?

File: src/psm/src/ir_solver.h [hunk 3]
```diff
@@ -62,7 +62,7 @@ class IRNetwork;
 class IRSolver
 {
  public:
-  using Voltage = float;
+  using Voltage = double;
   using Current = double;
   using Power = float;
```
Comment (maliberty) [inline src/psm/src/ir_solver.h:67]: Any reason to stick with float for Power?

**Pull request merged**